src/zkevm_specs/evm/execution/begin_tx.py

```diff
@@ -53,15 +54,21 @@ def begin_tx(instruction: Instruction):
     instruction.constrain_zero(instruction.add_account_to_access_list(tx_id, tx_caller_address))
     instruction.constrain_zero(instruction.add_account_to_access_list(tx_id, tx_callee_address))
 
+    account_value_rlc = instruction.account_read(tx_caller_address, AccountFieldTag.Balance)
```

Comment:
I think you can just reuse the `account_write` lookup that's already done in `transfer_with_gas_fee` because that also gives you the previous balance so it's the same lookup. So if you make `transfer_with_gas_fee` return the prev balance of the caller should be a slightly more efficient way to get this value.

Reply:
Could be as it saves this extra lookup, I just want to keep the changes less as possible, let me try digging into the transfer_with_gas_fee soon.

Reply:
Fixed, now it's a kind of post-checking of invalid_tx, as it writes the balance first and then check if the balance is not enough, but it do works.